source/common/router/config_impl.cc

```diff
@@ -933,10 +948,12 @@ VirtualHostImpl::VirtualHostImpl(const envoy::api::v2::route::VirtualHost& virtu
 
     if (validate_clusters) {
       routes_.back()->validateClusters(factory_context.clusterManager());
-      if (!routes_.back()->shadowPolicy().cluster().empty()) {
-        if (!factory_context.clusterManager().get(routes_.back()->shadowPolicy().cluster())) {
-          throw EnvoyException(fmt::format("route: unknown shadow cluster '{}'",
-                                           routes_.back()->shadowPolicy().cluster()));
+      for (const auto& shadow_policy : routes_.back()->shadowPolicies()) {
+        if (!shadow_policy->cluster().empty()) {
```

Comment:
What about the case where cluster is empty? If it can't happen due to config validations, I'd forego this check and perhaps just add an ASSERT

Reply:
yeah good call, there is a proto constraint that prevents the cluster from being empty. This was just a port from the previous logic, but moving this to an ASSERT would be a good improvement.